components/TipOfTheDay/index.tsx

```diff
@@ -1,11 +1,34 @@
+import { useEffect, useState } from "react"
 import styles from "./TipOfTheDay.module.css"
+import { TIPS } from "../../data/tipsOfTheDay"
 
-export const TipOfTheDay = () => {
-	return (
-		<div className={styles.card}>
-			<h2>Short tip title</h2>
-			<p>Longer explanation of tip</p>
-			<a href="#">Name of Link</a>
-		</div>
-	)
+interface ITip {
+	title: string
+	description: string
+	tipLink: string
+}
+
+export const TipOfTheDay: React.FC = () => {
+	const [randomIndex, setRandomIndex] = useState<number>(0)
+
+	useEffect(() => {
+		const TIPS_LENGTH = TIPS.length
+		const randomNumber = Math.floor(Math.random() * TIPS_LENGTH)
+
+		setRandomIndex(randomNumber)
+	})
+
+	const createTipOfTheDay = () => {
+		const tipIndex: ITip = TIPS[randomIndex]
+		return (
+			<>
+				<h2>{tipIndex.title}</h2>
+				<p>{tipIndex.description}</p>
+				<a target="_blank" rel="noreferrer" href={tipIndex.tipLink}>
+					Read More
+				</a>
+			</>
+		)
+	}
+	return <div className={styles.card}>{createTipOfTheDay()}</div>
```

Comment:
Calling functions inside of the component leads to flashing when the component renders on screen. If you try refreshing the page you'll see that the tip flashes a couple of times as things are causing it to render more than once.

It's better to call the function and store it in a variable, then pass only the variable into the component.